### ISSUE TITLE
Silverstripe image-shortcodes are picked up/shown as linkable in-page anchors

### ISSUE THREAD

**Issue by micschk** [silverstripe/silverstripe-cms#2837]
Images are inserted as shortcode in the editor:
```[image src="/assets/image.jpg" id="164" ... ]```

The id="164" part is then picked up and shown as on-page linkable anchor in the selector of the anchor-link modal in the CMS. (When selected, an anchor link gets inserted as `<a href="[sitetree_link,id=1]#164">...</a>`).

<img width="605" alt="Screenshot 2023-02-15 at 12 53 29" src="https://user-images.githubusercontent.com/1005986/219020465-1c2226ed-dd1c-456c-ac8c-641a802b1368.png">

One option could be to fix the regex itself at [`SiteTree::getAnchorsOnPage()`](https://github.com/silverstripe/silverstripe-cms/blob/77b5bfa497d5026ffc806f1071a6c5d04d65bfb4/code/Model/SiteTree.php#L3337) (NOTE: duplicated in [Elemental](https://github.com/silverstripe/silverstripe-elemental/pull/911/files#diff-f09cbb896dbc8d5589cdf7f33a628f570eabef6fd3ec6d06d2f3944a3b83570bR694)). 

But probably a better fix would be to process any shortcodes before filtering through the regex, as the result of shortcodes could theoretically contain linkable anchors as well?

## Steps to reproduce

1. Add an image (not a file link) to a WYSIWYG field
2. Save the page
3. Add link > anchor on a page. You will see the ID of the image in the anchor dropdown field.

## PRs

- https://github.com/silverstripe/silverstripe-cms/pull/3113

**Comment by michalkleiner** [comment]
> But probably a better fix would be to process any shortcodes before filtering through the regex, as the result of shortcodes could theoretically contain linkable anchors as well?

Yes, that would definitely be the case in our projects where shortcodes can provide markup that may contain links and anchors, too. At the same time it might be tricky since the content may not be directly visible if one went to check it manually since the WYSIWYG would still just have the shortcode.

**Comment by GuySartorelli** [comment]
I think processing shortcodes could be problematic, especially at large scale websites, and especially if custom shortcodes are added which take more time to process.

It might be better instead to _remove_ shortcodes prior to running the regex over the content. This means the regex doesn't have to get super complicated and we don't risk slowing down the CMS for complex shortcode processing.

**Comment by johannesx75** [comment]
Ok, I've made a [PR](https://github.com/silverstripe/silverstripe-cms/pull/3113).

In order to remove the shortcodes without breaking to much content, I'm using `extractTags` from the `ShortcodeParser`.
So we are not actually parsing / executing the shortcodes, but still use the same logik to remove them.

**Comment by GuySartorelli** [comment]
PR merged. Will be automatically tagged by GitHub Actions.
